include/atomics/receiverCadmium.hpp

```diff
@@ -28,85 +37,133 @@
+*/
+struct receiver_defs{
+    struct out : public out_port<message_t> {
+    };
+    struct in : public in_port<message_t> {
     };
-   
-    template<typename TIME>
-    class Receiver{
-        using defs=Receiver_defs; // putting definitions in context
-        public:
-            //Parameters to be overwriten when instantiating the atomic model
-            TIME   preparationTime;
-            // default constructor
-            Receiver() noexcept{
-              preparationTime  = TIME("00:00:10");
-              state.ackNum    = 0;
-              state.sending     = false;
-            }
+};
+
+/** 
+* The Receiver class receives message and sends out acknowledge.
+*/   
+template<typename TIME>
+class Receiver{
+    using defs=receiver_defs; // putting definitions in context
+    public:
+        //Parameters to be overwriten when instantiating the atomic model
+        TIME   PREPARATION_TIME;
+        
+	/** 
+	* Constructor for Receiver class.
+	* Initializes the delay constant and state structure.
+	*/
+        Receiver() noexcept{
+            PREPARATION_TIME  = TIME("00:00:10");
+            state.ack_num      = 0;
+            state.sending     = false;
+        }
+            
+        /**
+	* Structure that holds acknowledge number and receiver state.
+	*/
+        struct state_type{
+            int ack_num;
+            bool sending;
+        }; 
+        state_type state;
             
-            // state definition
-            struct state_type{
-              int ackNum;
-              bool sending;
-            }; 
-            state_type state;
-            // ports definition
-            using input_ports=std::tuple<typename defs::in>;
-            using output_ports=std::tuple<typename defs::out>;
+	// definition of different ports
+        using input_ports=std::tuple<typename defs::in>;
+        using output_ports=std::tuple<typename defs::out>;
 
-            // internal transition
-            void internal_transition() {
-              state.sending = false; 
-            }
+        /**
+	* Function that performs the internal transition.
+        * It turns off the receiver sending state.
+	*/
+        void internal_transition() {
+            state.sending = false; 
+        }
 
-            // external transition
-            void external_transition(TIME e, typename make_message_bags<input_ports>::type mbs) { 
-              if(get_messages<typename defs::in>(mbs).size()>1) assert(false && "one message per time uniti");
-              for(const auto &x : get_messages<typename defs::in>(mbs)){
-                state.ackNum = static_cast < int > (x.value);
+        /**
+	* Function that performs external transition.
+	* Retrieves the messages: if the number of messages
+	* is more than 1, it asserts that only one message is
+        * expected per time unit. It then sets the acknowledge
+	* to the message value and state to sending.
+	* @param e time variable
```

Comment:
It will look good if you type `@param` this way, `@param[in]` for input, also mentioned in slides

Reply:
Done